pkg/serverless/metrics/metric.go

```diff
@@ -57,8 +57,12 @@ func (m *MetricConfig) GetMultipleEndpoints() (map[string][]string, error) {
 
 // NewServer returns a running DogStatsD server
 func (m *MetricDogStatsD) NewServer(demux aggregator.Demultiplexer) (dogstatsdServer.Component, error) {
```

Comment:
You should be able to change the type here from `dogstatsdServer.Component` to `ServerlessDogstatsd` which will avoid the type cast in `Stop` below